### PR TITLE
feat(events): emit per-test timing in test results

### PR DESCRIPTION
**Pull request by codeman9**
Emit a `test-case-result` pipeline event for every individual test case (passed, failed, skipped) with its duration. Previously only `TestFailureEvent` carried per-test timing; passing and skipped tests were reduced to aggregate counters in `TestProgressEvent`, discarding the duration that xcodebuild already provides.

The new `TestCaseResultEvent` is emitted from both `xcodebuild-event-parser` and `swift-testing-event-parser`, collected in `XcodebuildRunState`, and rendered as a per-test timing breakdown before the summary line. The event carries `suite`, `test`, `status`, and `durationMs`.

Per-test output is opt-in via the `showTestResults` session default or `XCODEBUILDMCP_SHOW_TEST_RESULTS` env var. When disabled (the default), only the aggregate summary line is shown.

Example output when enabled:

```
Test Results:
  ✅ Suite/testA (0.001s)
  ❌ Suite/testB (0.002s)
  ⏭️ Suite/testC

✅ 2 tests passed, 1 skipped (⏱️ 3.2s)
```

All 1446 tests pass (6 new). Verified against a real iOS project (201 tests).

### CODE REVIEW

File: src/utils/swift-testing-event-parser.ts [hunk 1]
```diff
@@ -103,6 +123,7 @@ export function createSwiftTestingEventParser(
       const increment = stResult.caseCount ?? 1;
       completedCount += increment;
       failedCount += increment;
+      emitTestCaseResult(stResult);
       emitTestProgress();
```
[the remark this answers is not in the document]
Comment (codeman9) [reply src/utils/swift-testing-event-parser.ts:127]: *(Responding via kiro-cli on behalf of @codeman9)*

This is by design. Parameterized Swift tests are reported by xcodebuild as a single aggregate line, e.g.:

```
✔ Test "Parameterized test" with 3 test cases passed after 0.001 seconds.
```

We don't have individual case names or durations — just the group total. Emitting 3 identical events with the same name and duration would be misleading. The single `test-case-result` event accurately represents what xcodebuild provides. The `caseCount` increment on `test-progress` is correct for the aggregate counter, which is a separate concern from the per-test result event.

File: src/utils/renderers/cli-text-renderer.ts [hunk 2]
```diff
@@ -188,6 +193,11 @@ function createCliTextProcessor(options: CliTextProcessorOptions): PipelineRende
           break;
         }
 
+        case 'test-case-result': {
+          collectedTestCaseResults.push(event);
+          break;
+        }
```
[the remark this answers is not in the document]
Comment (codeman9) [reply src/utils/renderers/cli-text-renderer.ts:201]: *(Responding via kiro-cli on behalf of @codeman9)*

Good catch. Fixed in fdda57f5 — collection is now gated on `showTestResults`, matching the existing `suppressWarnings` pattern for `compiler-warning` events.

**Comment by codeman9** [comment]
@cameroncooke Warden seems to be failing on a get secrets step. Not sure if there is anything I need to do.

**Comment by cameroncooke** [comment]
Hi @codeman9 — first, thank you for filing this and for the original implementation work. The shape you proposed (a typed `TestCaseResultEvent` emitted by the parser, collected by run state, gated behind a `showTestResults` flag, formatted into a `Test Results:` section before the summary) was the right design, and the new PR follows that same shape end-to-end.

Unfortunately this PR can't be merged as-is. The blocker is purely mechanical and not a reflection on the change itself: the PR targets the legacy `PipelineEvent` streaming model, which has since been replaced on `main` by the `DomainFragment` + `RenderItem` architecture. `src/types/pipeline-events.ts` no longer exists, and the parser/run-state/renderer files your patches touch (`xcodebuild-event-parser.ts`, `xcodebuild-run-state.ts`, the legacy event-builder layer) have been refactored or removed. A clean rebase is no longer possible — the conflicts can't be resolved without effectively rewriting the patch.

I've reimplemented the feature from scratch against current `main` (PR #374), preserving your original behavior:
- `TestCaseResultFragment` in the new domain-fragments union (replacing the legacy `TestCaseResultEvent`)
- xcodebuild + Swift Testing parsing emits the fragment for passed / failed / skipped cases, with optional duration
- run state collects results; renderer gates collection on the flag and writes a `Test Results:` block before the summary
- icon set, duration formatting, and suite/test joining match your original byte-for-byte
- the parameterized-Swift-Testing edge case (one aggregate entry per group, since xcodebuild does not emit per-case names) is preserved as a known limitation

Two intentional differences from your original PR:

1. **Per-test data is now always emitted in structured/JSON output** as a new `testCases` field on `xcodebuildmcp.output.test-result`. This is a small but useful extension over the original — JSON consumers (e.g. agents, CI integrations) can rely on the data without needing to flip a flag. Text rendering remains opt-in.
2. **The flag is renamed `showTestTiming` and lives on top-level runtime config** (with `XCODEBUILDMCP_SHOW_TEST_TIMING=1` env var) instead of being a session default. This is a better fit for our config layering: the flag controls renderer behavior across CLI, JSON, and MCP runtimes uniformly, and session defaults are reserved for build/test invocation parameters.

The new PR credits you in the changelog as the originator (`#339 by @codeman9`).

Going to close this in favor of #374. Thanks again for the contribution — the feature is shipping, just under a different commit hash.
